renderdoc/driver/vulkan/vk_shaderdebug.cpp

```diff
@@ -4714,6 +4812,13 @@ ShaderDebugTrace *VulkanReplay::DebugPixel(uint32_t eventId, uint32_t x, uint32_
       continue;
     }
 
+    // if we're looking for a specific view, ignore hits from the wrong view
+    if(useViewIndex)
+    {
+      if(hit->view != view)
+        continue;
+    }
```

Comment:
I think this should use the same logic as below which prioritises getting a hit over the selected hit. E.g. for primitives we'll debug a different primitive if there are no hits from the selected one.